### PR TITLE
Updated commit hook to prevent from adding excessive includes when de…

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
…aling with boost, essentially will ignore suggested includes coming from boost.

Hey @JensWehner what do you think of this?

### CODE REVIEW

**Comment by JensWehner** [comment]
I do not understand it. What does it do?

**Comment by JoshuaSBrown** [comment]
1. It updates the license date to the current year, of every file you change
2. It calls include-what-you-use on every file you changed

* I have removed any includes that are automatically pulled in from boost, because include-what-you-use goes a little crazy with boost includes. 

3. It automatically calls clang-format on all the files you changed. 

You simply have to make the script executable and place it in the repo in .git/hooks with the file name pre-commit. 

Any time you call commit it will run.

**Comment by JensWehner** [comment]
do you want to wait till you can set a environment variable?

File: pre-commit [hunk 1]
```diff
@@ -14,16 +14,50 @@
...
+  ##############################################################################
+
   echo "Updating Copyright date in "$file
   sed -i 's/Copyright \(.*\)-.* The/Copyright \1-2019 The/g' $file
   echo "Applying clang style formatting to file "$file
+
+  ##############################################################################
+  # Updating Includes
+  ##############################################################################
+  data=$(include-what-you-use $file 2>&1 )
```
Comment (junghans) [inline pre-commit:82]:
```suggestion
if [[ ! ${DONT_USE_INCLUDE_WHAT_YOU_USE} ]]; then
    data=$(include-what-you-use $file 2>&1 )
 fi
```

**Review by junghans — CHANGES_REQUESTED** [review]

**Comment by JoshuaSBrown** [comment]
@junghans hey I fixed this.  Take a look.

**Pull request merged**